### ISSUE TITLE
Blank dialog on Linux/Wayland? Upgrade to Thunderbird 128.3.2

### ISSUE THREAD

**Issue by Wielding** [kewisch/quickmove-extension#158]
I have updated the extension for Thunderbird 128 so that it now is enabled on my installation.  However, when I use the keyboard shortcut or press the "Quick Move" button when Thunderbird is maximized, most of the time I am presented with a completely empty pop-up.  If I keep pressing the button eventually I get a usable pop-up, but that is rare.

When I say maximized I am refererring to it taking up an entire screen in my tiling window manager which is sway.

If I open another application next to Thunderbird so that Thunderbird is re-tilted to take up only half of my screen. The pop-up shows succesfully every time.  If I close the new window to let Thunderbird once again take up the entire screen the pop-up stops working again.

It also starts working again if I make Thunderbird a floating window and shrink the width so it as a guess it might possibly related to some CSS media queries?

This was not an issue in Thunderbird 115.

**Comment by 127001** [comment]
Not sure if I should open a new ticket as well or follow this one. I too have an empty Quick Move dialog, every time. But unlike @Wielding, I never get contents in there. 

Ubuntu 24.04
Thunderbird 128.0esr (64
Quick Folder Move 2.9.0

**Comment by 127001** [comment]
Just tried that, but for me that doesn't work. And compared to you, I get nothing in the overlay.

![Screenshot from 2024-07-18 12-00-25](https://github.com/user-attachments/assets/9371eb10-6b57-4812-8b6c-b2be04adb333)

**Comment by wondering639** [comment]
had the same issue. creating a new profile (also implies installing the extenstion again there) fixed this issue for me. I am now facing https://github.com/kewisch/quickmove-extension/issues/161 though ;-)

**Comment by dbonner** [comment]
Creating a new Thunderbird 128 profile did not fix this problem for me.  I still get a blank pop-up.

**Comment by l33tllama** [comment]
I'm having this issue as well with Thunderbird 128, Quick Folder Move 2.9.0 on Ubuntu 24.04. Just shows a blank box.

**Comment by hkclark** [comment]
I'm getting this too.  When I click the Quick Move button (or use the keyboard shortcut) I get the blank popup as others have mention.  In the console I get the following exception:
```
Uncaught (in promise) TypeError: accountMap[folder.accountId] is undefined
    fromList moz-extension://0ac9dd00-52d9-4364-97ef-3cd71a01314c/common/foldernode.js:58
    load moz-extension://0ac9dd00-52d9-4364-97ef-3cd71a01314c/popup/popup.js:101
    async* moz-extension://0ac9dd00-52d9-4364-97ef-3cd71a01314c/popup/popup.js:194
foldernode.js:58:24
    fromList moz-extension://0ac9dd00-52d9-4364-97ef-3cd71a01314c/common/foldernode.js:58
    load moz-extension://0ac9dd00-52d9-4364-97ef-3cd71a01314c/popup/popup.js:101
    InterpretGeneratorResume self-hosted:1417
    AsyncFunctionNext self-hosted:804
    (Async: async)
    <anonymous> moz-extension://0ac9dd00-52d9-4364-97ef-3cd71a01314c/popup/popup.js:194
```
Like @exmosis I can get it to display something if I select "All Folders" but if I then select a folder it just spins and never moves the email (which seems to be a different issue).

**Comment by dochego** [comment]
Similar to @exmosis: Switching the Default Folders config from "Recent folders" to "All folders" makes it work for me (moves emails, lets me change folders).
If I switch back to "Recent folders" it fails again (with any number of folders set). "All folders" works fine.

Ubuntu 22.04
Thunderbird 128.0.1esr (64-bit)
Quick Folder Move 2.9.0

**Comment by kewisch** [comment]
Fixed with Quick Folder Move 2.9.2 which is pending review.

**Comment by fwinkl** [comment]
I am still observing this issue of a blank folder list. I am on Thunderbird 128.1.0esr and Quick Folder Move 2.9.2. When clicking on the move button, I see this message in the error console:
```
Promise rejected after context unloaded: Actor 'Conduits' destroyed before query 'RuntimeMessage' was resolved
```
and the popup looks like in the screenshots above. Changing from recent to all or specific folders doesn't help either.

**Comment by Wielding** [comment]
I still have the issue as well with 2.9.2 and 128.1.0 on Linux.

**Comment by kewisch** [comment]
If you look at your recent folder list, does anything pop out? Some uncommon type of folder? You can see it via right click and move/copy, this will show the built in Thunderbird recent folders.

**Comment by MWWorks** [comment]
I am having the same issue - is it maybe Linux specific? Also on Ubuntu 24.04, I have latest 2.9.2 and tried:

- New profile
- Removing all other extensions
- Downloading TB from site vs using Snap in Ubuntu
- Switching to/from all/recent/specific folders
- Keyboard shortcut vs toolbar button

My behaviour is a fully blank overlay - although if I keep trying, it works randomly, maybe 1 time in 10?

My folders are a few levels deep in some places but I don't see any unusual characters or other weirdness.

**Comment by fwinkl** [comment]
Nothing unusual in my built-in recent folder list. Just 7 regular IMAP folders.

But I can confirm the random behavior reported by @MWWorks. For me it works about 1/20 times. The rest of the time the dialog box is empty.

**Comment by l33tllama** [comment]
With 2.9.2 I am still experiencing the issue, the pop up box is still blank I'm afraid :( Ubuntu 24.04

**Comment by codexico** [comment]
Same here, 2.9.2 and Thunderbird 128.1.0esr on Ubuntu 24.04.
What I can add is that  I tried the behaviors above and it opened with content just one time, after resizing the window, but never again.

**Comment by MWWorks** [comment]
I can confirm the same behaviour as @codexico - I tried with the QuickFolders extension and also with the Signature Switch, which also adds a button to the top bar. It seems to matter that the Quick Move button is on the left, rather than the right.

Quick Move on RHS:
![image](https://github.com/user-attachments/assets/8568edfd-a6fe-4cc2-8a9f-be89268b8e5d)

Quick Move on LHS seems to consistently work:
![image](https://github.com/user-attachments/assets/a0af1e82-49f3-40f1-84c1-24412eaa17dd)

As per @codexico, the  small button in the mail toolbar does not work:
![image](https://github.com/user-attachments/assets/46b7fd2b-2f56-43dc-a8ad-989a14ea22e1)

I thought it was when the Quick move popup is not entirely within the bounds of the main TB window, but that doesn't align with @codexico's second screenshot.

It seems the keyboard shortcut for Go to folder, invokes the top bar button and thus works as expected; but the shortcut for Move messages invokes the smaller button and so does not work.
 
Hopefully this is some useful info progress :)

**Comment by kewisch** [comment]
This sounds very promising, hope to get back soon. Thanks for investigating!

**Comment by kewisch** [comment]
I wasn't able to reproduce with placement. Is there anyone not on Ubuntu/Linux where this is failing?

**Comment by exmosis** [comment]
Things are working much better for me with 2.9.2 - thanks for the fix. I haven't seen the issues above, but am on Linux 22 - will be interesting to see if it changes when the upgrade to 24 comes out officially soon.

I might be able to test it on Mac later on too, if time.

**Comment by MWWorks** [comment]
By adjusting the button position as above, this is largely resolved for me, apart from the move keyboard shortcut:

- Goto folder shortcut - invokes top bar button - works OK
- Move message (when multiple messages selected) - invokes top bar button - works OK
- Move message (when one message selected) - invokes message panel button - blank (as it extends outside TB main window)

Ie if you are stuck reproducing/solving this, is it possible to change the button target of the keyboard shortcut? That would provide a workaround for all cases.

**Comment by alessandromannini** [comment]
Same blank popup here on Windows 11 (updated till today) and v.2.9.2

![immagine](https://github.com/user-attachments/assets/d9603f42-c474-4bc0-9635-5ecdfbbdaef4)

Some other hints:
1) WHEN I CLICK on the Quick Move button on the top bar I get this console error:

`[Exception... "Component returned failure code: 0x80004005 (NS_ERROR_FAILURE) [nsIMsgFolder.server]"  nsresult: "0x80004005 (NS_ERROR_FAILURE)"  location: "JS frame :: resource:///modules/ExtensionAccounts.sys.mjs :: convert :: line 423"  data: no] ExtensionAccounts.sys.mjs:423:20
Uncaught (in promise) Error: An unexpected error occurred undefined
Promise rejected after context unloaded: Actor 'Conduits' destroyed before query 'RuntimeMessage' was resolved
popup.js:162
    unload moz-extension://078bdb44-08a9-45b6-9d6f-320e2bb76b73/popup/popup.js:162
`

2) WHEN I OPEN the right click menu the Move To tree is empty:
![immagine](https://github.com/user-attachments/assets/34c34ebc-a0e4-47cc-abf5-60191923546b)


Thanks
Alessandro

**Comment by kewisch** [comment]
> [Exception... "Component returned failure code: 0x80004005 (NS_ERROR_FAILURE) [nsIMsgFolder.server]"  nsresult: "0x80004005 (NS_ERROR_FAILURE)"  location: "JS frame :: resource:///modules/ExtensionAccounts.sys.mjs :: convert :: line 423"  data: no] ExtensionAccounts.sys.mjs:423:20

Aha, this sounds like a problem that might trigger it, but could unfortunately point to an issue in Thunderbird. It looks like the folder being shown doesn't have an incoming server associated to it, for some reason. Before you change anything in your settings please make note/screenshots of it so we can help identify the issue in Thunderbird. Can you then go through the "Server Settings" tab of each account and see if there are any invalid settings? What types of accounts do you have (email, news, etc) ?

Note to self: https://searchfox.org/comm-esr128/source/mail/components/extensions/ExtensionAccounts.sys.mjs#423

**Comment by alessandromannini** [comment]
> Aha, this sounds like a problem that might trigger it, but could unfortunately point to an issue in Thunderbird. It looks like the folder being shown doesn't have an incoming server associated to it, for some reason. Before you change anything in your settings please make note/screenshots of it so we can help identify the issue in Thunderbird. Can you then go through the "Server Settings" tab of each account and see if there are any invalid settings? What types of accounts do you have (email, news, etc) ?
> 
> Note to self: https://searchfox.org/comm-esr128/source/mail/components/extensions/ExtensionAccounts.sys.mjs#423

Found an old (and forgotten) News account, deleted it and **ALL** re-started to work! Both TB standard move and QFM!
With previous TB version I didn't have this problem.

Many thanks!!!
Alessandro

**Comment by mirenradia** [comment]
I'm getting this issue with the Thunderbird flatpak (128.1.1esr) on Fedora 40 KDE.

As mentioned by a few others in this issue, I observe the behaviour where it randomly works a small proportion of the time.

I had a look at the console and I can't consistently see anything. Occasionally, I see this error message after click away from the blank dialog box.

```
Promise rejected after context unloaded: Actor 'Conduits' destroyed before query 'RuntimeMessage' was resolved
popup.js:162
    unload moz-extension://6cd47e63-c15f-4c7f-9228-429c8a1cd72a/popup/popup.js:162
```

**Comment by exmosis** [comment]
Just upgraded to Ubuntu 24 (from 22) and can confirm that it has now _stopped_ working for me as per others above, v2.9.2 in TB 128.1.0esr (64-bit). I just get a blank window now.

Also confirming that it works fine on latest Thunderbird on Mac. Stupidly, I didn't get a similar log under Ubuntu 22 for comparison.

Here's the log from starting Thunderbird up from the command line: [thunderbird_quickmove_log.txt](https://github.com/user-attachments/files/16840147/thunderbird_quickmove_log.txt) - not sure if there's anything useful in there, or why it's complaining about the appmenu-gtk-module not loading. I have appmenu-gtk2-module and appmenu-gtk3-module installed.

I've tried removing all my RSS feeds and an experimental mail account, but no luck yet. Will carry on checking over account-related errors though.

**Comment by JalilArfaoui** [comment]
Same issue on Thunderbird 128.1.0esr on Linux 6.10.6 (NixOS 24.05) with Gnome 46 (Wayland). 

![image](https://github.com/user-attachments/assets/7e2929b4-3c61-43c6-bd16-cc6a448d6fb9)

**Comment by neilzone** [comment]
Using Thunderbird 128.1.1esr on Debian, with QM 2.9.2, the pane is blank when I invoke it using a keyboard shortcut.

It is also blank when invoked from the menu bar, but I've made it work by moving the menu item for QM to the left of the search bar.

**Comment by kewisch** [comment]
This seems to be a Linux specific issue based on the comments. I don't have a Linux machine ready and waiting, so I might not get to this immediately. I'm suspecting though it might be a Thunderbird bug, because I am not doing anything fancy that would be platform specific (aside from keyboard shortcuts). The fully blank popup is also telling for that.

If someone can try to debug this further in an exploratory fashion I'd be grateful!

**Comment by neilzone** [comment]
> This seems to be a Linux specific issue based on the comments.  I don't have a Linux machine ready and waiting, so I might not get to this immediately. I'm suspecting though it might be a Thunderbird bug, because I am not doing anything fancy that would be platform specific (aside from keyboard shortcuts). The fully blank popup is also telling for that.

I did some testing last night, and, looking for a workaround, I found [mmy](https://addons.thunderbird.net/en-US/thunderbird/addon/move-messages/?src=search). This adopts a similar approach to moving messages, using a pop-up pane.

This add-on has the same problem QM, which makes me think that, as you say, this is a Thunderbird bug.

(However, with mmy, the placement of the pane means that, if I move the pane to the left of the search bar, it does work sufficiently to be usable.)

> If someone can try to debug this further in an exploratory fashion I'd be grateful!

I'd be happy to try, but I don't really know where to start. Any suggestions of things i could read to learn, or any tips, would be most welcome.

**Comment by dochego** [comment]
Linux, KDE Neon Plasma 6
I can confirm this bug when I run thunderbird in a Wayland session. When I start thunderbird in a X session, quick-move works fine.
So this seems to be not only Linux specific, but Wayland specific, which is another indicator that it's a bug in Thunderbird.

**Comment by kewisch** [comment]
You are not the first to experience this, here is the upstream bug:

https://bugzilla.mozilla.org/show_bug.cgi?id=1916112

**Comment by Josef-Friedrich** [comment]
Linux Ubuntu 24.04 Thunderbird 128.2.0esr Wayland

![Screenshot from 2024-09-15 16-31-15](https://github.com/user-attachments/assets/b6d7fe5f-b25d-4ce6-a665-14265d68d18b)

**Comment by jobisoft** [comment]
I landed a fix in Thunderbird Daily (2 days ago), which aims to fix the empty popus (without using `widget.wayland.use-move-to-rect`).

Can you confirm the popups working again?

**Comment by mirenradia** [comment]
> I landed a fix in Thunderbird Daily (2 days ago, [Bug 1905622](https://bugzilla.mozilla.org/show_bug.cgi?id=1905622)), which aims to fix the empty popus (without using `widget.wayland.use-move-to-rect`).
> 
> Can you confirm the popups working again?

Will this fix make it to a patch release of Thunderbird 128?

**Comment by jobisoft** [comment]
> Will this fix make it to a patch release of Thunderbird 128?

If it actually fixes the issue for a larger group, then yes.

**Comment by fwinkl** [comment]
I tried testing on daily 132.0a1 (2024-09-25) but it seems the extension doesn't work at all. Fails to load with the exception:

```
[Exception... "File error: Not found"  nsresult: "0x80520012 (NS_ERROR_FILE_NOT_FOUND)"  location: "JS frame :: jar:file:///home/fwinkl/.thunderbird/etewtpx1.daily/extensions/quickmove@mozilla.kewis.ch.xpi!/api/quickmove.js :: <TOP_LEVEL> :: line 7"  data: no] quickmove.js:7:34
    <anonymous> jar:file:///home/fwinkl/.thunderbird/etewtpx1.daily/extensions/quickmove@mozilla.kewis.ch.xpi!/api/quickmove.js:7
    asyncLoaded resource://gre/modules/ExtensionCommon.sys.mjs:1796
    (Async: promise callback)
    asyncLoadModule resource://gre/modules/ExtensionCommon.sys.mjs:1790
    asyncLoadModule resource://gre/modules/ExtensionCommon.sys.mjs:2022
    asyncGetAPI resource://gre/modules/ExtensionCommon.sys.mjs:1734
    onStartup resource://gre/modules/ExtensionCommon.sys.mjs:1532
    onStartup resource://gre/modules/ExtensionCommon.sys.mjs:2000
    map self-hosted:175
    onStartup resource://gre/modules/ExtensionCommon.sys.mjs:2000
    apiManager resource://gre/modules/ExtensionParent.sys.mjs:99
[...]
    (Async: EventListener.handleEvent)
    promiseInitialized chrome://mozapps/content/extensions/aboutaddons.js:4218
    <anonymous> chrome://mozapps/content/extensions/aboutaddons.js:4217
```

**Comment by axelsimon** [comment]
Just to report that moving the button to the left of the Thunderbird built-in "Search… <kb>CTRL</kb> + <kb>K</kb>" field makes Quickmove work again. Strange. Hope it helps.

**Comment by olaulau** [comment]
> Just to report that moving the button to the left of the Thunderbird built-in "Search… CTRL + K" field makes Quickmove work again. Strange. Hope it helps.

the click on the button yes
but not the keystroke shurtcut

**Comment by Josef-Friedrich** [comment]
> Just to report that moving the button to the left of the Thunderbird built-in "Search… CTRL + K" field makes Quickmove work again. Strange. Hope it helps.

I can confirm this

If the menu looks like this example, it is working

```
--------------
| Quick move |
|--------------------|
|                    |
|                    |
|--------------------|
```

not working

```
        --------------
        | Quick move |
|--------------------|
|                    |
|                    |
|--------------------|
```

**Comment by 127001** [comment]
I'd like to add another situation where the keyboard shortcut does work. When I select multiple mails and use the keyboard shortcut, the menu opens up. 
Not sure if it is possible to remove the quick move button (temporarily) the message view, but I think the plugin would then work again. Taking in account that I also moved the general quick move button left of the search.

**Comment by olaulau** [comment]
> > Just to report that moving the button to the left of the Thunderbird built-in "Search… CTRL + K" field makes Quickmove work again. Strange. Hope it helps.
> 
> the click on the button yes but not the keystroke shortcut

nevermind, this seems to work even with the keystroke when button is on the left side.

**Comment by glennergeerts** [comment]
I can confirm that: 

- moving the Quick Move button to the top left solves the issue; the folders are displayed again. 
- when using the Quick Move button in a mail pane (so not the top left one), the pop-up is empty again. 
- when using the shortcut when a single mail is selected the pop-up is empty as well. 
- when the shortcut is used when multiple mails are selected it works again (it uses the top left menu in this case).

Using the shortcut to move a single mail is the main use case for me, so it would be good to find a similar workaround as moving the top button, for the time being. I don't see any possible workaround at the moment, however. I would be happy to test or try to debug something on my system if useful. I'm on TB 128.3.1esr on Fedora 40 (Wayland) using Quick Folder Move v2.9.2 

Thanks for the efforts on this add-in btw, it's really useful!

**Comment by martin8032** [comment]
I also have the issue with the blank dialogue (Quick Move 2.9.2, Thunderbird 128.3.1esr (64-Bit), Ubuntu 24.10).

As far as I can see, this happens only when the Thunderbird window is wider than 1400 px.

**Comment by fwinkl** [comment]
Yes, I can confirm that the problem is fixed in 128.3.2 for me. Nice!

**Comment by kewisch** [comment]
Sweet, thanks for confirming. I'll leave this open for a while longer so folks can easily see the solution when it is pinned and then close it off. Thank you everyone for testing and hunting for solutions, and John for fixing this in 128!
